core/types/system_tx.go

```diff
@@ -0,0 +1,104 @@
+package types
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/rlp"
+)
+
+type SystemTx struct {
+	ChainID *big.Int
+	To      *common.Address // system contract
+	Nonce   uint64          // nonce
+	Data    []byte          // calldata
+	V, R, S *big.Int        // signature values
+}
+
+func (tx *SystemTx) txType() byte { return SystemTxType }
+
+func (tx *SystemTx) copy() TxData {
+	cpy := &SystemTx{
+		ChainID: new(big.Int),
+		Nonce:   tx.Nonce,
+		To:      copyAddressPtr(tx.To),
+		Data:    common.CopyBytes(tx.Data),
+		V:       new(big.Int),
+		R:       new(big.Int),
+		S:       new(big.Int),
+	}
+
+	if tx.ChainID != nil {
+		cpy.ChainID.Set(tx.ChainID)
+	}
+	if tx.V != nil {
+		cpy.V.Set(tx.V)
+	}
+	if tx.R != nil {
+		cpy.R.Set(tx.R)
+	}
+	if tx.S != nil {
+		cpy.S.Set(tx.S)
+	}
+
+	return cpy
+}
+
+func (tx *SystemTx) chainID() *big.Int {
+	return tx.ChainID
+}
+
+func (tx *SystemTx) accessList() AccessList {
+	return nil
+}
+
+func (tx *SystemTx) data() []byte {
+	return tx.Data
+}
+
+func (tx *SystemTx) gas() uint64 {
+	return 0
```

Comment:
This is the execution gas, if you set it to `0` it will lead to out-of-gas

Reply:
yeah, was just thinking about this.

Reply:
So what is a good value here? Hard to guess.

Reply:
but this comes from the config, no? duh.

Reply:
added a Gas field, it must be specified -- it is not reasonable to hardcode anything.